kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -2109,18 +2104,18 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         if (mDoFMasterFixedSet.size() > 0) {
             // NOTE: dofs are assumed to be numbered consecutively
             const auto it_dof_begin = BaseType::mDofSet.begin();
-            #pragma omp parallel for
-            for(int k = 0; k < static_cast<int>(mDoFToSolveSystemSize); ++k) {
-                auto it_dof = it_dof_begin + k;
-                if (k < static_cast<int>(BaseType::mEquationSystemSize)) {
+
+            IndexPartition<std::size_t>(mDoFToSolveSystemSize).for_each([&](std::size_t Index){
+                auto it_dof = it_dof_begin + Index;
+                if (Index < BaseType::mEquationSystemSize) {
```

Comment:
seems like you need to explicitly capture `mEquationSystemSize` as it is protected and hence cannot be accessed in the lambda without being captured

I am confused as to whether this is a bug in Gcc 4.8 or not

Reply:
How should I capture it again ??  `[&]` includes everything right ?

Reply:
I don't think `&` captures the class instance itself (`this`)
Didn't check the docs though

Reply:
Ohh I see... I will try with `this`

Reply:
`&` captures everything in the scope where the lambda is declared. To capture the class itself you need to specify the `this`.

Reply:
Still the error is same ,
`>::mEquationSystemSize' is protected
     unsigned int mEquationSystemSize; /// Number of degrees of freedom of the problem to be solve
                  ^
In file included from /__w/Kratos/Kratos/kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:44:0,
                 from /__w/Kratos/Kratos/build/Custom/kratos/CMakeFiles/KratosCore.dir/Unity/unity_8_cxx.cxx:8:
/__w/Kratos/Kratos/kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h:2110:27: error: within this context
                 if (Index < BaseType::mEquationSystemSize)`

Am I making some mistake again?

Reply:
hmmmm
can you try this?
```suggestion
                if (Index < this->mEquationSystemSize) {
```

Reply:
Working now . Thanks :)